week7/cooper_client_react/01_cooper_calculator.md

```diff
@@ -0,0 +1,398 @@
+
+We are going to setup the acceptance testing first. Like we did in the BMI Calculation project, we are going to use `jest-puppeteer`.
+
+We need to configure it now. Run:
+```shell
+$ touch jest-puppeteer.config.js
+$ touch jest.config.js
+```
+And add this to them:
+
+```js
+// jest.config.js
+module.exports = {
+    verbose: true,
+    preset: "jest-puppeteer",
+    testRegex: ".feature\\.js$"
+}; 
+```
+
+```js
+// jest-puppeteer.config.js
+module.exports = {
+    launch: {
+        headless: false,
+        slowMo: 10,
+        devtools: true,
+        timeout: 100000,
+        args: ['--disable-setuid-sandbox',
+            '--no-sandbox',
+            '--ignore-certificate-errors',
+            "--disable-popup-blocking",
+            "--disable-infobars",
+            '--disable-web-security']
+    },
+    browserContext: 'default',
+
+    server: {
+        command: `PORT=3001 BROWSER=none npm run start`,
+        port: 3001,
+        launchTimeout: 4000,
+    },
+    
+} 
+```
+
+Now you need to add a script to the `package.json` to be able to run these tests
+
+```js
+"scripts": {
+  "start": "react-scripts start",
+  "build": "react-scripts build",
+  "test": "react-scripts test",
+  "eject": "react-scripts eject",
+  "test:features": "jest -c jest.config.js"
+}
+```
+It is time to add our first test now. Run:
+
+`$ mkdir src/__features__`
+
+`$ touch src/__features__/UserCanGetCooperCalculationResult.feature.js`
+
+Add these tests to the new feature file:
+
+```js
+describe('Cooper Client', async () => {
+  beforeAll(async () => {
+    await page.goto('http://localhost:3000');
+  });
+
+  beforeEach(async () => {
+    await page.reload();
+  })
+  
+
+  describe('calculates successfully', async () => {
+    beforeEach(async () => {
+      await page.type('input[id="distance"]', '1000')
+      await page.select('select[id="gender"]', 'female')
+      await page.type('input[id="age"]', '23')
+    })
+
+    it('displays age', async () => {
+      await expect(page).toMatch('23 y/o')
+    })
+
+    it('displays gender', async () => {
+      await expect(page).toMatch('female')
+    })
+
+    it('displays distance', async () => {
+      await expect(page).toMatch('running 1000 meters')
+    })
+
+    it('displays result', async () => {
+      await expect(page).toMatch('Result: Poor')
+    })
+  })
+})
+```
+
+It is a pretty straigth forward test. The user fills in the distance, selects their gender and types in their age. When all of these inputs have values we should get a response with their result. 
+
+***"++Remember to commit often++"***
+
+Lets start with cleaning up `App.js` aka the App component. When that is done we want to add the inputs and selectors that we need to make this feature test to go green.
+The App component should look like this:
+
+```js
+import React, { Component } from 'react';
+import './App.css';
+
+class App extends Component {
+ render() {
+    return (
+      <div className="App">
```

Comment:
We don't want to keep the scaffolded class names

Reply:
removed